src/freenet/crypt/KeyGenUtils.java

```diff
@@ -36,6 +36,10 @@
  */
 public final class KeyGenUtils {
 
+    static {
+        java.security.Security.insertProviderAt(new BouncyCastleProvider(), 1);
```

Comment:
Alternatively instead of modifying security we could save the instance in a property.